### PR TITLE
Add initial procedural crosswalks and related behavior.

### PR DESCRIPTION
**Pull request by gio-tempo**
- Enable building procedural crosswalks, crosswalk intersections, and crosswalk connectors (ie. the "ZoneShape" between two crosswalk intersections).

- Add "vehicle to pedestrian" and "pedestrian to vehicle" reactive yield behaviors.

- Tweak the intersection periods for 4-way and T-intersections with traffic lights and stop signs to allow crosswalk lanes to be open while vehicle lanes are open, in order to increase the overall throughput at intersections.  The yield logic handles the negotiation between vehicles and pedestrians as they traverse the intersections.

- Added mechanism to enforce mandatory stops at stop signs even when the intersection "period" logic has lanes currently open at stop signs.  You can set `LowerMinStopSignRestTime` and `UpperMinStopSignRestTime` in MassTrafficSettings.h to set the range of acceptable time to wait at a stop sign before proceeding.

- Enable partial cross-traffic at 4-way and T stop sign intersections.  "Partial" just means that during intersection "periods" that allow left turns, no cross-traffic is allowed.  But, in all other intersection "periods" cross-traffic *is* allowed.

- Enable crosswalk signaling on traffic lights for all updated intersection "period" logic.

### CODE REVIEW

File: TempoAgents/Source/TempoAgentsEditor/Private/TempoRoadLaneGraphSubsystem.cpp [hunk 1]
```diff
@@ -728,6 +748,8 @@ bool UTempoRoadLaneGraphSubsystem::TryGenerateAndRegisterZoneShapeComponentsForC
 
 bool UTempoRoadLaneGraphSubsystem::TryGenerateAndRegisterZoneShapeComponentsForCrosswalkIntersections(AActor& IntersectionQueryActor) const
 {
+	// FEditorScriptExecutionGuard ScriptGuard;
```
Comment (gio-tempo) [inline TempoAgents/Source/TempoAgentsEditor/Private/TempoRoadLaneGraphSubsystem.cpp:751]: Remove this.

File: External/MassCrowd/Source/MassCrowd/Public/MassCrowdTypes.h [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+	/**
+	 * Index of an associated waiting area. This will be valid for branching lanes leading
+	 * to a crossing that requires agent to wait for incoming lane to open.
+	 */
+	int32 WaitAreaIndex = INDEX_NONE;
+
+	int32 NumEntitiesOnLane = 0;
+
+	// TODO:  Hook-up "Lead" distance as well.
```
Comment (gio-tempo) [inline External/MassCrowd/Source/MassCrowd/Public/MassCrowdTypes.h:61]: Remove this.

File: External/Traffic/Source/MassTraffic/Private/MassTrafficLaneChange.cpp [hunk 3]
```diff
@@ -1151,6 +1192,109 @@ bool ShouldPerformReactiveYieldAtIntersection(
...
+		
+		if (!ensureMsgf(CrowdTrackingLaneData != nullptr, TEXT("Must get valid CrowdTrackingLaneData in IsDownstreamCrosswalkLaneClear.  TestDownstreamCrosswalkLane.Index: %d."), TestDownstreamCrosswalkLane.Index))
+		{
+			// Since we're in an error state, assume it's clear to keep traffic flowing.
+			return true;
+		}
+
+		if (CrowdTrackingLaneData->NumEntitiesOnLane > 0)
+		{
+			if (!ensureMsgf(CrowdTrackingLaneData->TailEntityNormalizedDistanceAlongLane.IsSet(), TEXT("CrowdTrackingLaneData has Entities on the lane, but no TailEntityNormalizedDistanceAlongLane value in IsDownstreamCrosswalkLaneClear.  TestDownstreamCrosswalkLane.Index: %d."), TestDownstreamCrosswalkLane.Index))
```
Comment (gio-tempo) [inline External/Traffic/Source/MassTraffic/Private/MassTrafficLaneChange.cpp:1233]: Ensure `LeadEntityNormalizedDistanceAlongLane.IsSet()` here, too.

File: External/Traffic/Source/MassTraffic/Private/MassTrafficIntersectionSpawnDataGenerator.cpp [hunk 4]
```diff
@@ -469,121 +457,337 @@ void UMassTrafficIntersectionSpawnDataGenerator::Generate(UObject& QueryOwner, T
...
+							Period.VehicleLanes.Append(VehicleTrafficLanes_Right_To_OppositeAndRight);
+						}
+						
+						// Period -
+						//		Vehicles - This side to opposite side and its right.  Opposite side to this side and its right.
+						//		Pedestrians - None
+						{
+							FMassTrafficPeriod& Period = IntersectionFragment.AddPeriod(StandardMinimumTrafficGoSeconds);
+
+							Period.VehicleLanes.Append(VehicleTrafficLanes_Left_To_OppositeAndRight);
```
[the remark this answers is not in the document]
Comment (gio-tempo) [reply External/Traffic/Source/MassTraffic/Private/MassTrafficIntersectionSpawnDataGenerator.cpp]: Added missing code for 4-way stop sign periods.  (This also resolved the duplicate period.)

**Review by pete-tempo — APPROVED** [review]
Reviewed over VC

**Pull request merged**